backport of https://github.com/taisei-project/koishi/commit/4adb18203b
fix koishi' coroutine stack allocation: needs MAP_STACK

Index: external/koishi/src/stack_alloc.c
--- external/koishi/src/stack_alloc.c.orig
+++ external/koishi/src/stack_alloc.c
@@ -39,7 +39,7 @@ static inline void *alloc_stack_mem(size_t size) {
 #if defined KOISHI_HAVE_WIN32API
 	return VirtualAlloc(NULL, size, MEM_COMMIT, PAGE_READWRITE);
 #elif defined KOISHI_HAVE_MMAP
-	return mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | KOISHI_MAP_ANONYMOUS, -1, 0);
+	return mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_STACK | MAP_PRIVATE | KOISHI_MAP_ANONYMOUS, -1, 0);
 #elif defined KOISHI_HAVE_ALIGNED_ALLOC
 	return aligned_alloc(koishi_util_page_size(), size);
 #elif defined KOISHI_HAVE_POSIX_MEMALIGN
